### ISSUE TITLE
Exception when creating test module

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#3877]
Since PR #3867 was introduced I just observed an unexpected error when adding a test module for where I should be able to. Turns out that we now have a RCW disconnection:

```
2018-04-02 10:53:31.0086;WARN-2.1.6665.42525;Rubberduck.UI.Command.AddTestModuleCommand;System.Runtime.InteropServices.InvalidComObjectException: COM object that has been separated from its underlying RCW cannot be used.
   at System.StubHelpers.StubHelpers.GetCOMIPFromRCW(Object objSrc, IntPtr pCPCMD, IntPtr& ppTarget, Boolean& pfNeedsRelease)
   at Microsoft.Vbe.Interop.VBProjectClass.get_Protection()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBProject.get_Protection() in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBProject.cs:line 48
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBProject.get_VBComponents() in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBProject.cs:line 46
   at Rubberduck.UI.Command.AddTestModuleCommand.OnExecute(Object parameter) in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.Core\UI\Command\AddTestModuleCommand.cs:line 172;System.Runtime.InteropServices.InvalidComObjectException: COM object that has been separated from its underlying RCW cannot be used.
   at System.StubHelpers.StubHelpers.GetCOMIPFromRCW(Object objSrc, IntPtr pCPCMD, IntPtr& ppTarget, Boolean& pfNeedsRelease)
   at Microsoft.Vbe.Interop.VBProjectClass.get_Protection()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBProject.get_Protection() in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBProject.cs:line 48
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBProject.get_VBComponents() in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBProject.cs:line 46
   at Rubberduck.UI.Command.AddTestModuleCommand.OnExecute(Object parameter) in C:\GitHub\Rubberduck\Rubberduck\Rubberduck.Core\UI\Command\AddTestModuleCommand.cs:line 172
```

We are getting the error because we are trying to get the [`VBComponents` in a `using`](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Core/UI/Command/AddTestModuleCommand.cs#L172). This shouldn't be a cause for such error....

**Comment by MDoerner** [comment]
Looking at the call stack from the exception, I would say that the `VBProject` passed es parameter has been disposed already. Why this is the case will require more investigation.

**Comment by bclothier** [comment]
Actually, I think the problem comes from the call to `GetProject()`. We use that when the parameter is null as per the [line #155-156](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Core/UI/Command/AddTestModuleCommand.cs#L155):

```
        private IVBProject GetProject()
        {
            using (var activeProject = _vbe.ActiveVBProject)
            {
                if (!activeProject.IsWrappingNullReference)
                {
                    return activeProject;
                }
            }

            using (var projects = _vbe.VBProjects)
            {
                return projects.Count == 1
                    ? projects[1]
                    : new VBProject(null);
            }
        }
```

Since it's inside a `using`, it gets disposed once returned. Not sure why the `parameter` is sometimes null, though.

**Comment by MDoerner** [comment]
I think the `parameter` is left `null` or not set to a module or project in case it is supposed to work on the active project.

The using statement on the `activeProject` probably is the culprit here. It should really not be there.

**Comment by Vogel612** [comment]
on that note: it should be `projects[0]`, shouldn't it?

**Comment by bclothier** [comment]
No, not a safe assumption. It is possible for host add-ins to load earlier and thus be the first "project" when we normally mean the active VB project which could be any where.

**Comment by bclothier** [comment]
Fixed in the PR #3876
